src/java/org/apache/cassandra/db/compaction/CompactionTask.java

```diff
@@ -205,17 +207,24 @@ public boolean apply(SSTableReader sstable)
                             }
                         }
 
-                        // don't replace old sstables yet, as we need to mark the compaction finished in the system table
+                        readyToFinish = true;
                         newSStables = writer.finish();
                     } catch (Exception e) {
-                        throw new CompactionException(taskIdLoggerMsg, ssTableLoggerMsg.toString(), e);
+                        CompactionException exception = new CompactionException(taskIdLoggerMsg, ssTableLoggerMsg.toString(), e);
+                        if (readyToFinish && e.getSuppressed() != null && e.getSuppressed().length != 0)
```

Comment:
Can a suppressed exception not be null but have a size of 0?

Reply:
And we have confirmed that the only time a suppressed exception is present is when the close statement fails?